src/chapters/more-on-functions/receiving-functions.rst

```diff
@@ -0,0 +1,210 @@
+
+We can write a logging function that relies on a function parameter to determine the logging destination.
+
+A Simple Logger
+^^^^^^^^^^^^^^^
+
+.. admonition:: Example
+
+   The ``logError`` function outputs a standardized error message to a location determined by the parameter ``logger``.
+
+   .. sourcecode:: js
+      :linenos:
+   
+      let fileLogger = function(msg) {
+
+         // Put the message in a file
+
+      }
+
+      function logError(msg, logger) {
+         let errorMsg = 'ERROR: ' + msg;
+         logger(errorMsg);
+      }
+
+      logError('Something broke!', fileLogger);
+
+Let's examine this example in more detail.
+
+There are three main program components:
+
+#. Lines 1-5 define ``fileLogger``, which takes a string argument, ``msg``. We have not discussed writing to a file, but Node.js is capable of doing so. 
+#. Lines 7-10 define ``logError``. The first parameter is the message to be logged. The second parameter is the logging function that will do the work of sending the message somewhere. ``logError`` doesn't know the details of how the message will be logged. It simply formats the message, and calls ``logger``.
+#. Line 12 logs an error using the ``fileLogger``.
+
+This is the flow of execution:
+
+#. ``logError`` is called, with a message and the logging function ``fileLogger`` passed as arguments.
+#. ``logError`` runs, passing the constructed message to ``logger``, which refers to ``fileLogger``.
+#. ``fileLogger`` executes, sending the message to a file.
+
+A More Complex Logger
+^^^^^^^^^^^^^^^^^^^^^
+
+This example can be made even more powerful by enabling multiple loggers.
+
+.. admonition:: Example
+
+   The call to ``logError`` will log the message to both the console and a file.
+
+   .. sourcecode:: js
+      :linenos:
+   
+      let fileLogger = function(msg) {
+
+         // Put the message in a file
+
+      }
+
+      let consoleLogger = function(msg) {
+      
+         console.log(msg);
+      
+      }
+
+      function logError(msg, loggers) {
+
+         let errorMsg = 'ERROR: ' + msg;
+
+         for (let i = 0; i < loggers.length; i++) {
+            logger[i](errorMsg);
+         }
+
+      }   
+
+      logError('Something broke!', [fileLogger, consoleLogger]);
+
+The main change to the program is that ``logError`` now accepts an *array* of functions. It loops through the array, calling each logger with the message string.
+
+As with the validation example, these programs separate behaviors in a way that makes the code more flexible. To add or remove a logging destination, we can simply change the way that we call ``logError``. The code *inside* ``logError`` doesn't know about how each logging function does it's job. It is concerned merely with creating the message string and passing it to the logger(s).
```

Comment:
Grammar: Remove the 'about' from "...doesn't know about how each logging function..."

Personal preference: "only" instead of "merely".